src/runtime_src/core/common/api/native_profile.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (C) 2016-2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "native_profile.h"
+#include "core/common/module_loader.h"
+#include "core/common/utils.h"
+#include "core/common/dlfcn.h"
+
+// Anonymous namespace for helper functions
+namespace {
+
+static std::string full_name(const char* type, const char* function)
+{
+  if (type == nullptr) return function ;
+  
+  std::string combined = type ;
+  combined += "::" ;
+  combined += function ;
+  return combined ;
+}
+
+} // end anonymous namespace
+
+namespace xdp {
+namespace native {
+
+bool load_xdp_native()
```

Comment:
This was where I think you should drop both xdp and _native, making it `xdp::native::load()`   Same for other namespaced methods.  This of course is not  important, just idiomatic.

Reply:
Dropped xdp and native from the names of the functions in the latest push.